shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/view/StudyPlanViewState.kt

```diff
@@ -0,0 +1,36 @@
+package org.hyperskill.app.study_plan.view
+
+sealed interface StudyPlanViewState {
+    object Idle : StudyPlanViewState
+
+    object Loading : StudyPlanViewState
+
+    object Error : StudyPlanViewState
+
+    data class Content(
+        val sections: List<Section>
+    ) : StudyPlanViewState
+
+    data class Section(
+        val id: Long,
+        val title: String,
+        val subtitle: String,
+        val content: SectionContent
+    )
+
+    sealed interface SectionContent {
+        object Collapsed : SectionContent
+
+        object Loading : SectionContent
+
+        object Error : SectionContent
+
+        data class Content(
+            val sectionItems: List<SectionItem>
+        ) : SectionContent
```

Comment:
should we add logic for load more section's items?

Reply:
Yes, I'll do it in future PRs. At the moment we loading all section's activities without pagination.